### PR TITLE
support channel last in BatchNorm*d

### PR DESCRIPTION
**Pull request by iclementine**
### PR types

Function optimization

### PR changes

APIs

### Describe

1. Add support for channel last data layout for BatchNorm1d, BatchNorm2d, BatchNorm3d.
2. Fix a bug in batch_norm op's cuda kernel. (conv_op.h and batch_norm_op.h have a bunch of functions with the same name in the same namespace, but with slightly different functionalities, which cause a error. This problem is then solved by extracting the common functions into a new header file and included it into conv_op.h and  batch_norm_op.h)

### CODE REVIEW

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by qingqing01 — COMMENTED** [review]
如Hi上的讨论，确认下，cuDNN BatchNorm对于NHWC的支持，是否可以通过设置stride，而不需要Trans Channel做到支持？(这样的好处省去了计算前后的2次转置)

**Comment by iclementine** [comment]
> 如何Hi上的讨论，确认下，cuDNN BatchNorm对于NHWC的支持，是否可以通过设置stride，而不需要Trans Channel做到支持？(这样的好处省去了计算前后的2次转置)

但是 https://github.com/PaddlePaddle/Paddle/pull/21090 这个 PR 中说的是手动转置更快。
